cmd/cli/user.go

```diff
@@ -49,10 +59,81 @@ func deleteUser(database moira.Database, user string) error {
 	if err != nil {
 		return err
 	}
+
 	for _, contactID := range contactIDs {
 		if err := database.RemoveContact(contactID); err != nil {
 			return err
 		}
 	}
+
 	return nil
 }
+
+func handleCleanup(logger moira.Logger, database moira.Database, config cleanupConfig) error {
+	var users []string
+
+	reader := json.NewDecoder(bufio.NewReader(os.Stdin))
+
+	if err := reader.Decode(&users); err != nil {
+		return err
+	}
+
+	return usersCleanup(logger, database, users, config)
+}
+
+func usersCleanup(logger moira.Logger, database moira.Database, users []string, config cleanupConfig) error {
+	usersMap := make(map[string]bool, len(users)+len(config.Whitelist))
+
+	if config.HandlingAnonymous {
+		config.Whitelist = append(config.Whitelist, "")
+	}
+
+	for _, user := range append(users, config.Whitelist...) {
+		usersMap[user] = true
+	}
+
+	contacts, err := database.GetAllContacts()
+	if err != nil {
+		return err
+	}
+
+	var usersNotFound []string
+
+	for _, contact := range contacts {
+		if !usersMap[contact.User] {
+			usersNotFound = append(usersNotFound, contact.User)
+		}
+	}
+
+	for _, user := range usersNotFound {
+		if config.Delete {
+			if err = deleteUser(database, user); err != nil {
+				break
+			}
+		} else if err = offNotification(database, user); err != nil {
+			break
+		}
+
+		logger.Info("User cleaned:", user)
+	}
+
+	return err
+}
+
+func offNotification(database moira.Database, user string) error {
+	subscriptionIDs, err := database.GetUserSubscriptionIDs(user)
+	if err != nil {
+		return err
+	}
+
+	subscriptions, err := database.GetSubscriptions(subscriptionIDs)
+	if err != nil {
+		return err
+	}
+
+	for _, subscription := range subscriptions {
+		subscription.Enabled = false
```

Comment:
Unfortunately, sometimes it's just nil pointer

Reply:
BTW: i think we can also flush already scheduled notifications here (https://github.com/moira-alert/moira/blob/3fa28a373f082c49a0c1dc6b6d512470d35bb496/database/redis/notification.go#L54)